samples/browser/src/main/java/org/mozilla/samples/browser/request/SampleRequestInterceptor.kt

```diff
@@ -21,12 +21,22 @@ class SampleRequestInterceptor(val context: Context) : RequestInterceptor {
         engineSession: EngineSession,
         uri: String,
         hasUserGesture: Boolean,
-        isSameDomain: Boolean
+        isSameDomain: Boolean,
+        isAppInitiated: Boolean
     ): InterceptionResponse? {
         return when (uri) {
             "sample:about" -> InterceptionResponse.Content("<h1>I am the sample browser</h1>")
-            else -> context.components.appLinksInterceptor.onLoadRequest(
-                engineSession, uri, hasUserGesture, isSameDomain)
+            else -> {
+                var response = context.components.appLinksInterceptor.onLoadRequest(
+                        engineSession, uri, hasUserGesture, isSameDomain, isAppInitiated)
+
+                if (response == null && !isAppInitiated) {
+                    response = context.components.webAppInterceptor.onLoadRequest(
```

Comment:
Looks like we're opening PWA (if the session exists) regardless what the user preference is.  Is this what we want?

Reply:
@rocketsroger currently this is not tied to any perference, yes. I personally think it would be good to let users enable / disable this behavior, but a preference was not mentioned in the original fenix issue mozilla-mobile/fenix#5772.